miden-vm/src/cli/prove.rs

```diff
@@ -115,10 +115,9 @@ impl ProveCmd {
 
         let input_data = InputFile::read(&self.input_file, &self.program_file)?;
         let (program, source_manager) = match ext.as_str() {
-            "masp" => (
-                get_masp_program(&self.program_file)?,
-                Arc::new(DefaultSourceManager::default()) as Arc<dyn SourceManager>,
-            ),
+            "masp" => {
+                (get_masp_program(&self.program_file)?, DefaultSourceManager::default_arc_dyn())
```

Comment:
nit: The `DefaultHost::default()` method already instantiates the default source manager, so we could remove this (and the call to `DefaultHost.with_source_manager()`).

This also applies to the other cli files